interfaces/builtin/x11.go

```diff
@@ -149,10 +150,26 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+var x11ConnectedPlugMount = []osutil.MountEntry{{
+	Name:    "/var/lib/snapd/hostfs/tmp/.X11-unix/",
+	Dir:     "/tmp/.X11-unix/",
+	Options: []string{"bind"},
+}}
```

Comment:
This is clearly correct for the `implicitSystemPermanentSlot(slot)` case, but not otherwise.  In that case, the source probably needs to be `/var/lib/snapd/hostfs/tmp/snap.${slot_snap}/tmp/.X11-unix`.  That probably requires a `MountConnectedPlug` implementaqtion rather than the default `commonInterface` code.  Ideally this would use some kind of helper to determine the slot snap's private tmp directory path, to avoid getting out of sync with snap-confine.

I think it would be worth making this a read-only bind mount too, to prevent the plug snap from modifying the directory contents.

Reply:
yes mount read-only is much appreciated here.

Side note: `abstractions/X` is currently broken in AppArmor 3.0+ because socket doesn't [have write access](abstractions/X) which is surprisingly needed in apparmor semantics.

Reply:
I think this is done.

CC @alexmurray and @jdstrand about apparmor X abstraction and `w+` permission on sockets? Will the read-only bind mount pose a problem here?

Reply:
read-only mount should work, flatpak does it and other sandbox frameworks too.

Reply:
abstractions/X will be [fixed in next apparmor 3 release](https://gitlab.com/apparmor/apparmor/-/commit/0cb35fda84a6ace742d9da3a7630a0dcc6ffae9d)